modules/apps/static/portal-configuration/portal-configuration-persistence-impl/src/main/java/com/liferay/portal/configuration/persistence/internal/upgrade/v1_0_0/UpgradeConfigurationPid.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.configuration.persistence.internal.upgrade.v1_0_0;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.io.unsync.UnsyncByteArrayInputStream;
+import com.liferay.portal.kernel.io.unsync.UnsyncByteArrayOutputStream;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.Dictionary;
+
+import org.apache.felix.cm.file.ConfigurationHandler;
+
+/**
+ * @author Sam Ziemer
+ */
+public class UpgradeConfigurationPid extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (hasTable("Configuration_")) {
+			try (PreparedStatement preparedStatement1 =
+					connection.prepareStatement("select * from Configuration_");
+				PreparedStatement preparedStatement2 =
+					AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+						connection,
+						"update Configuration_ set configurationId = ?, " +
+							"dictionary = ? where configurationId = ?");
+				ResultSet resultSet = preparedStatement1.executeQuery()) {
+
+				while (resultSet.next()) {
+					String dictionaryString = resultSet.getString("dictionary");
+
+					Dictionary<String, String> dictionary =
+						ConfigurationHandler.read(
+							new UnsyncByteArrayInputStream(
+								dictionaryString.getBytes(StringPool.UTF8)));
+
+					String serviceFactoryPid = dictionary.get(
+						"service.factoryPid");
+
+					if (serviceFactoryPid == null) {
+						continue;
+					}
+
+					String currentConfigurationId = resultSet.getString(
+						"configurationId");
+
+					int serviceFactoryPidEndIndex = serviceFactoryPid.length();
+
+					String updatedConfigurationId = StringBundler.concat(
+						serviceFactoryPid, CharPool.TILDE,
+						currentConfigurationId.substring(
+							serviceFactoryPidEndIndex + 1));
+
+					dictionary.put("service.pid", updatedConfigurationId);
+
+					String felixFileInstallFilename = dictionary.get(
+						"felix.fileinstall.filename");
+
+					if (felixFileInstallFilename != null) {
```

Comment:
@SamZiemer The logic inside this if block is wrong. You should not change the file install name in dictionary, as this will cause the file name stored in database is not consistent with what's in the file system.